### PR TITLE
Add -compress option for compressing binary output.

### PR DESCRIPTION
**Pull request by Slipyx**
This adds a -compress option for compressing binary output files. It uses STM compression by sverx for Tilemaps and PSGaiden compression by maxim for Tile data. I included the original compressor sources from BMP2Tile and modified them for statically building. Using -compress also implies -binary and prints a warning message if it wasn't also specified. Also added some verbosity by printing the current filename being processed as well as file size results when using compression.

Comments / suggestions are welcome. Thanks.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,11 +3,17 @@ project(png2tile)
 
 find_package(PNG REQUIRED)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
```
Comment (yuv422) [inline CMakeLists.txt]: Just wondering why this change was required. Does it introduce some gnu only functions?

Comment (Slipyx) [reply CMakeLists.txt]: Oh, it's not really required, it's just what I usually use since gcc itself defaults to the gnu standard over the regular ones. Same with the C flags. I left it in from when I added some other flags I was using locally. If you want I can change it back to c++11. I can also remove the cflags.

Comment (yuv422) [reply CMakeLists.txt]: Yes lets revert back to c++11
